internal/providers/terraform/aws/db_instance.go

```diff
@@ -192,6 +191,36 @@ func NewDBInstance(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
 		})
 	}
 
+	if !d.IsEmpty("backup_retention_period") || (u != nil && !u.IsEmpty("additional_backup_storage_gb")) {
+		var backupStorageGB *decimal.Decimal
+		if u != nil && !u.IsEmpty("additional_backup_storage_gb") {
+			backupStorageGB = decimalPtr(decimal.NewFromInt(u.Get("additional_backup_storage_gb").Int()))
+		}
+
+		backupStorageDbEngine := "Any"
+		if strings.HasPrefix(*databaseEngine, "Aurora") {
```

Comment:
```suggestion
		if databaseEngine != nil && strings.HasPrefix(*databaseEngine, "Aurora") {
```
Should we add a nil ptr check here?